crates/stages/src/stages/hashing_storage.rs

```diff
@@ -68,11 +68,10 @@ impl<DB: Database> Stage<DB> for StorageHashingStage {
                         .walk(first_key)?
                         .take(self.commit_threshold as usize)
                         .map(|res| {
-                            res.map(|(address, mut slot)| {
+                            res.map(|(address, slot)| {
                                 // both account address and storage slot key are hashed for merkle
                                 // tree.
-                                slot.key = keccak256(slot.key);
-                                (keccak256(address), slot)
+                                ((keccak256(address), keccak256(slot.key)), slot.value)
                             })
                         })
                         .collect::<Result<BTreeMap<_, _>, _>>()?;
```

Comment:
Yeah, this is a bug. 

Maybe just changing `.collect::<Result<BTreeMap<_, _>, _>>()?;` to `.collect::<Result<Vec<_>, _>>()?;` is little bit better?

Reply:
we don't need to collect here at all right? can we not pre-seek to the equivalent `storage.next()`index w/o collecting?

Reply:
I believe it's needed. The `collect` into a `BTreeMap` removes duplicate entries, and orders the batch, and should make insertions into the DB faster.